### ISSUE TITLE
Bug: [ban-types] Should suggest object, not Record<string, never>

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#5947]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.4&jsx=true&sourceType=module&code=CYUwxgNghgTiAEEQBd4CsBc8CuA7A1rgPYDuuA3AFBKoC28AvPAN7wB0HAFGvFAM4sAvgEp4gqkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AIwEMnajTUMkRNGgB7aJHBgAviGlA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
declare let j: unknown;
let m = { ...(j as {}) };
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/ban-types": "error",
  },
};
```

### Expected Result

From @RyanCavanaugh in https://github.com/typescript-eslint/typescript-eslint/pull/5018:

> Now that `object` has been shipping in TS for several years, this seems like a safe suggestion in lieu of `Record`'s messy side effects

From @bradzacher in https://github.com/typescript-eslint/typescript-eslint/pull/5018#issuecomment-1132269029:

> The issue with `object` as a suggested replacement has always been that it's really hard use (https://github.com/microsoft/TypeScript/issues/21732).

#21732 -> github.com/Microsoft/TypeScript/pull/50666 has been shipped, so the major flaws mentioned should be gone now.

Proposal: let's suggest `object`:

```diff
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
+ If you want a type meaning "any object", you probably want `object` instead.
```

I'm unsure what to do about `{}`. On the one hand:

> Having typescript-eslint, out of the box, ban you from writing a type that TypeScript itself is inferring under totally normal operation, is awkward.

On the other hand, people do mistake it for "any object". 🤔 

### Actual Result

Right now we still recommend:

```plaintext
Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead. 
```

### Additional Info

Originally suggested by @RyanCavanaugh in https://github.com/typescript-eslint/typescript-eslint/pull/5018.

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.42.1` |
| `@typescript-eslint/parser` | `5.42.1` |
| `TypeScript` | `4.8.4` |
| `ESLint` | `8.15.0` |
| `node` | `web` |
